src/charm.py

```diff
@@ -91,6 +91,14 @@ class UPFOperatorCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to preform if we're removing the
```

Comment:
```suggestion
            # teardown code is necessary to perform if we're removing the
```

Reply:
I have no idea why, but I always make this typo :laughing: thanks.

Reply:
Fixed in https://github.com/canonical/sdcore-upf-operator/pull/52/commits/7083a54806325bb35a6545c17705460ff3a0856b